### ISSUE TITLE
[BUG]: v2 beta install fails

### ISSUE THREAD

**Issue by davidegraff** [chemprop/chemprop#566]
**Describe the bug**
Attemping to run `pip install .` with on the current V2 source fails

**Expected behavior**
The above command should run without error

**Error Stack Trace**
```
$ pip install .
Obtaining file:///Users/degraff/projects/chemprop
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build editable did not run successfully.
  │ exit code: 1
  ╰─> [15 lines of output]
      WARNING setuptools_scm._integration.setuptools pyproject.toml does not contain a tool.setuptools_scm section
      error: Multiple top-level packages discovered in a flat-layout: ['data', 'chemprop'].
      
      To avoid accidental inclusion of unwanted files or directories,
      setuptools will not proceed with this build.
      
      If you are trying to create a single distribution with multiple packages
      on purpose, you should not rely on automatic discovery.
      Instead, consider the following options:
      
      1. set up custom discovery (`find` directive with `include` or `exclude`)
      2. use a `src-layout`
      3. explicitly set `py_modules` or `packages` with a list of names
      
      To find more information, look for "package discovery" on setuptools docs.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build editable did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Comment by kevingreenman** [comment]
When I checkout this PR and install as follows,
```
conda create -n chemprop-clean-v2-test python=3.11
python -m pip install . 
```
I get an error while installing `torch-scatter` because `torch` hasn't been installed yet
```
Collecting torch-scatter (from chemprop==2.0.0b1)
  Using cached torch_scatter-2.1.2.tar.gz (108 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-m8hqneag/torch-scatter_8d8d0732cc1c4d438b6e225d597ea288/setup.py", line 8, in <module>
          import torch
      ModuleNotFoundError: No module named 'torch'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
```

When I install via 
```
conda create -n chemprop-clean-v2-test python=3.11
pip install torch
python -m pip install . 
```
I don't get any errors.

**Comment by kevingreenman** [comment]
Why are we not observing this behavior in our CI testing?

**Comment by davidegraff** [comment]
I believe because our CI is first installing the environment using [`environment.yml`](https://github.com/chemprop/chemprop/blob/master/environment.yml) ([see here](https://github.com/chemprop/chemprop/blob/v2/dev/.github/workflows/tests.yml#L26)) before installing the package. As a side note, I get issues attempting to install `torch_scatter` from PyPI and generally have to [install the precompiled binary](https://github.com/rusty1s/pytorch_scatter?tab=readme-ov-file#binaries). That is:
```
$ conda activate ENV
$ pip install [all packages except torch-scatter]
$ pip install torch-scatter -f [torch-scatter whl URL]
```
For what it's worth, I'm running into this issue on my mac, so it might not be an issue on Linux.

**Comment by davidegraff** [comment]
Also it looks like it's a general issue trying to install both `torch` and `torch_scatter` [at the same time](https://github.com/rusty1s/pytorch_scatter/issues/265)

**Comment by kevingreenman** [comment]
Fixed by #567
